File: solutions_5658571765186560_0/Python/arunpoudel/q4.py
for t in range (int (raw_input())):
    
    X, R, C = map (int, raw_input(). split())
    
    ans = True
    
    if X >= 7 :
        ans = False
    elif X > R and X > C:
        ans = False
    elif R * C % X is not 0 :
        ans = False
    elif (X + 1 ) // 2 > min (R, C):
        ans = False
    elif X in ( 1 , 2 , 3 ):
        ans = True
    elif X == 4 :
        ans = min (R, C)> 2
    elif X == 5 :
        ans = not (min (R, C) == 3  and max (R, C) == 5 )
    elif X == 6 :
        ans = min (R, C)> 3

    print 'Case #%d:' % (t + 1), 'GABRIEL'  if ans else  'RICHARD'

File: solutions_5658571765186560_1/java/Jonyzfu/Q4.java
package online;
import java.io.*;
import java.lang.Math;
import java.util.Scanner;


public class Q4 {
	public static void main(String[] args) throws Exception {
		Scanner in = new Scanner(new File("/Users/jonyzfu/Documents/D-large.in"));
		PrintWriter out = new PrintWriter(new FileWriter("/Users/jonyzfu/Documents/D-large.out"));
		int t = in.nextInt();
		for (int i = 1; i <= t; i++) {
			int x = in.nextInt();
			int r = in.nextInt();
			int c = in.nextInt();
			boolean res = omino(x, r, c);
			if (res) {
				out.println("Case #"+i+": GABRIEL");
			} else {
				out.println("Case #"+i+": RICHARD");
			}
			
		}
		in.close();
		out.close();
	}
	
	public static boolean omino(int x, int r, int c) {
		boolean res = true;
		if (x >= 7) {
			res = false;
		} else if (x > r && x > c) {
			res = false;
		} else if (r * c % x != 0) {
			res = false;
		} else if (Math.floor((double)(x + 1) / 2.0) > Math.min(r, c)) {
			res = false;
		} else if (x == 1 || x == 2 || x == 3) {
			res = true;
		} else if (x == 4) {
			res = Math.min(r, c) > 2;
		} else if (x == 5) {
			res = !(Math.min(r, c) == 3 && Math.max(r, c) == 5);
		} else if (x == 6) {
			res = Math.min(r, c) > 3;
		}
		return res;
	}
}